### ISSUE TITLE
Screenshots missing from metainfo.xml files

### ISSUE THREAD

**Issue by Eonfge** [freedoom/freedoom#1305]
The latest Flathub package can't use the official metainfo.xml because the images linked in the screenshots-secion are 404.

https://buildbot.flathub.org/#/builders/6/builds/96885

**Comment by mc776** [comment]
What does "the images linked in the screenshots-secion" mean? If we're talking about the links https://freedoom.github.io/screenshots.html[here] that was fixed in the past couple hours so it might need time to refresh.

What time zones are those timestamps for?

**Comment by Eonfge** [comment]
This file has a lists of screenshots

https://github.com/freedoom/freedoom/blob/v0.13.0/dist/io.github.freedoom.Phase2.metainfo.xml

But when I follow the first link...

https://freedoom.github.io/img/screenshots/phase2-gzdoom-map05-shotgun-zombies-0.13.0.png

I receive a 404, not found.

**Comment by unithecutie** [comment]
https://freedoom.github.io//img/screenshots/phase2-gzdoom-map05-shotgun-zombies-0.13.0.jpg

This one works.

**Comment by mc776** [comment]
It's too late to change the link, it's already part of the release.

I've re-uploaded those four screenshots.

**Comment by Eonfge** [comment]
Same with FreeDM and Phase 1 :disappointed: 

Sorry for bothering you with such a trivial error, but other distributions will also run into such problems. Over on Flathub, we have a series of backup images within the repository, because 5 years ago we didn't support webm images.

https://github.com/flathub/io.github.freedoom.Phase1/tree/master/images

Perhaps it would also be best to make the metainfo.xml screenshots part of your repository: `./dist/screenshots/phase1` could contain all the relevant screenshots, so they are also versioned.

**Comment by selliott512** [comment]
@Eonfge, certainly not too trivial. These XML files have been somewhat mysterious in that I don't know of a good way of testing them. I just test that they are well formed XML with valid URLs (at the time I did it). I didn't know that Flathub was consuming them. I thought they were for something like "Software" GUI front end to the repos in Linux distros. Your report reinforces the idea that they are important, and worth maintaining.

@mc776, there are 12 image URLs in question, 4 for each of the 3 IWADs. This is from the XML files in question:
```
https://freedoom.github.io/img/screenshots/freedm-gzdoom-dm01-banner-0.13.0.png
https://freedoom.github.io/img/screenshots/freedm-gzdoom-dm06-exit-0.13.0.png
https://freedoom.github.io/img/screenshots/freedm-gzdoom-dm21-banner-0.13.0.png
https://freedoom.github.io/img/screenshots/freedm-gzdoom-dm22-corner-0.13.0.png
https://freedoom.github.io/img/screenshots/phase1-gzdoom-e1m6-upstairs-0.13.0.png
https://freedoom.github.io/img/screenshots/phase1-gzdoom-e2m8-tripods-0.13.0.png
https://freedoom.github.io/img/screenshots/phase1-gzdoom-e3m1-trilobites-worms-0.13.0.png
https://freedoom.github.io/img/screenshots/phase1-gzdoom-e3m3-pain-lord-0.13.0.png
https://freedoom.github.io/img/screenshots/phase2-gzdoom-map05-shotgun-zombies-0.13.0.png
https://freedoom.github.io/img/screenshots/phase2-gzdoom-map11-serpentipede-0.13.0.png
https://freedoom.github.io/img/screenshots/phase2-gzdoom-map26-mushrooms-0.13.0.png
https://freedoom.github.io/img/screenshots/phase2-gzdoom-map32-matribite-hatchling-0.13.0.png
```
I'd be happy to help with the remaining 8 (the first 8 in the above list), but if you don't mind you can apply the same pngcrush process to them, and they'll be done consistently.

Maybe I should've explored more efficient image formats prior to the release. Maybe I was overly eager to match match what had been done for previously releases. Next time.

**Comment by selliott512** [comment]
Another thought - both the PNG and JPG could continue to exist to reduce traffic for those vising the Screenshots page while satisfying these tagged (unchangeable) XML files.

**Comment by Eonfge** [comment]
@selliott512 The files are a little cryptic, especially when you don't regularly package things for Linux.

The entire spec is here:
- https://freedesktop.org/software/appstream/docs/chap-Metadata.html

There is also a tool to check all these components: 
- https://github.com/flathub-infra/flatpak-builder-lint

But you can ignore most of it, since you already have mostly valid metainfo files. It's just the images who are missing. 

And yes, Flathub checks them to enforce some best practices. You can in fact skip release tags and screenshots altogether if you want, but Flathub tries to raise the bar so they check them before you can publish something. GNOME Software will just show a '?' if the image get lost for whatever reason.

**Comment by mc776** [comment]
> bESt prActiCes
If I were in charge of this project I would drop Flathub support altogether as a matter of principle.

I'm shrinking the re-uploaded images because using that much bandwidth for already-outdated images *using GZDoom's default blurriness* is unconscionable.

**Comment by mc776** [comment]
> Maybe I should

The problem is that this stuff is just so extremely sprawling with so much reduplicated information everywhere...

**Comment by selliott512** [comment]
I'm fine with Flathub in general, but Freedoom is not the best fit. Freedoom is not really software by itself, it's the artwork and game data that is then combined with a game engine.

It's surprisingly difficult to figure out which game engine [Freedoom: Phase 2](https://flathub.org/apps/io.github.freedoom.Phase2) on Flathub is using. After installing it and examining the install location it became clear that it was GZDoom. The Flathub entry should mention this fundamental fact.

IMHO Freedoom should just be a library / dependency on Flathub. Then people could create a `Freedoom: Phase 2 With GZDoom`  (notice that I added the game engine to allow for others) Flathub entry, or whatever they want.

Freedoom has a very permissive license, so ultimate people can do what they want.

**Comment by Eonfge** [comment]
> If I were in charge of this project I would drop Flathub support altogether as a matter of principle.

Whaha, well... I'm the guy maintaining both GZDoom and the Freedoom projects on Flathub... and I too don't consider the Freedoom's to be that important. The project wasn't even started by me, I adopted it since I already did GZDoom. As @selliott512 Put it, it's not really a stand-alone product.

The only reason I keep the Flatpaks around on Flathub, is because of its limited amount of work and because of the free publicity. By being included on Flathub with images and metadata, you're directly available on many distributions without any extra work.

You're upstream and your say it worth a lot to me. If you really want to abandon Flathub that it can be done. I'll move all existing users (~40,000) to GZDoom and add the files to the GZDoom project.

**Comment by selliott512** [comment]
@Eonfge Unless I'm confused this issue was closed because the problem because the screenshots have been fixed. I actually do think it's neat that Freedoom has that extra publicity, it's just structured a bit different than I think it should be. I understand how it is to inherit projects, and how they can be different than you would've done it.

**Comment by Eonfge** [comment]
I've prepared an updated package for Phase 2:
https://github.com/flathub/io.github.freedoom.Phase2/pull/11

I'll not be releasing it today since I've run out of hobby-time today, but I'll see if I can prepare the other two packages tomorrow. I'll then release them all three at the same time.

**Comment by Eonfge** [comment]
Issue got fixed. Thanks for your help

**Comment by selliott512** [comment]
@Eonfge I've had a chance to look at the three Flathub entries (Phase 1, Phase 2 and FreeDM) more closely. I like it. It seems to be true that:
* You are not self promoting.
* They link to the Freedoom website.
* They require minimum permissions with Phase 1 and Phase 1 not requiring any permissions.
* It works quite well - the user can play immediately without any configuration.

You probably had a colder reception on this thread than you deserved.

However I don't know right now that there is a good path to verification since the Freedoom project has little interest in taking on addition packaging, nor would they want to endorse something as being somehow official if it's not under Freedoom's GitHub account. But I will continue to learn about it.

You are welcome to do what you want with Freedoom, as I mentioned, and as you know, given its license, but if you don't mind it would help clarify things if the Flathub entries mentioned that GZDoom was chosen as a game engine, and that Freedoom itself is engine agnostic. That being said I think GZDoom is a fine choice for users that aren't particularly motivated to recreate the 1990s gaming experience. I play with GZDoom the most of any of the engines.

And finally, I'd like to better understand what the relationship is between Freedoom, and these three Flathub entries. How do you consume the `metainfo.xml` entries? Do you pull down the ZIP, or do you process the git tag? For those not familiar the files are:
```
dist/io.github.freedoom.FreeDM.metainfo.xml
dist/io.github.freedoom.Phase1.metainfo.xml
dist/io.github.freedoom.Phase2.metainfo.xml
```

How could the Freedoom project make things easier for you? Are the files correct as is?

How helpful would it be if Freedoom produced something like a Freedoom Flathub library that your three Flathub enries could then depend on? Would that help automate things for you? I'm not saying the rest of the Freedoom project would want to do this, but I'm exploring some options.

**Comment by Eonfge** [comment]
Hey, thanks for taking a closer look.

> You probably had a colder reception on this thread than you deserved.

Don't worry about it. Flathub still has some detractors and I understand that many open-source projects are not that engaged with the distribution on Linux. Most FLOSS game(-engines) are after all cross-platform and their primary target is Windows. Freedoom goes even one step further than that, and it doesn't even ship a software at all. 

As for the metadata entries; the FreeDesktop metainfo standard helps Linux desktop in listing age ratings, promotional images, and contact info. 

![Screenshot from 2024-02-02 08-45-27](https://github.com/freedoom/freedoom/assets/33983090/7ac75387-cca4-40ca-b60c-5dd82aebb038)

As for the build process, it's now as simple as it gets really. I don't actually 'recompile' the .wad since there is no need to. Plain assets don't benefit from strict compiler settings. Since I use the distribution-agnostic package with the .wad pre-build, I have to manually download the .desktop and .metainfo.xml This is totally fine since it's not much work, and it would be unreasonable to make Linux, Mac and Windows specific .zip files, just so you have the right metadata.

To make it a little clearer, I've added additional annotations the build process for Phase2:
https://github.com/flathub/io.github.freedoom.Phase2/blob/annotated/io.github.freedoom.Phase2.yaml

**Comment by Eonfge** [comment]
For comparison, here is the Phase 2 package from Fedora Linux, which uses `prboom` from 2008 to run. 

![Screenshot from 2024-02-02 09-06-08](https://github.com/freedoom/freedoom/assets/33983090/e9349f22-a9da-41c1-a578-3be0cdb90612)

Packages like these disappoint me because they use 15 year old, outdated technology to run a project that many volunteers put so much work in. I too despise the GZDoom's default fussy-textures... but at least GZDoom supports modern graphics libraries, controllers, and mods.

Last funny detail, the age-rating for Phase 2 is higher than for GZDoom. When I had to come up with the GZDoom age-rating, I looked at the modern PEGI rating and it's 16+... which makes sense. Doom in 1993 was the pinnacle op digital manslaughter but by 2024 standards it looks really quant and comedic.
